sdk/storage/storage-blob/src/BlobServiceClient.ts

```diff
@@ -173,6 +173,55 @@ export class BlobServiceClient extends StorageClient {
     });
   }
 
+  /**
+   * Iterates over containers under the specified account.
+   *
+   * @param {ServiceListContainersSegmentOptions} [options={}] Options to list containers(optional)
+   * @returns {AsyncIterableIterator<Models.ContainerItem>}
+   * @memberof BlobServiceClient
+   *
+   * @example
+   * for await (const container of blobServiceClient.listContainers()) {
+   *   console.log(`Container: ${container.name}`);
+   * }
+   *
+   * @example
+   * let iter1 = blobServiceClient.listContainers();
+   * let i = 1;
+   * for await (const container of iter1) {
+   *   console.log(`${i}: ${container.name}`);
+   *   i++;
+   * }
+   *
+   * @example
+   *  let iter2 = await blobServiceClient.listContainers();
+   *  i = 1;
+   *  let containerItem = await iter2.next();
+   *  do {
+   *    console.log(`Container ${i++}: ${containerItem.value.name}`);
+   *   containerItem = await iter2.next();
+   * } while (containerItem.value);
+   *
+   */
+  public async *listContainers(
```

Comment:
nit: move `*` to be with `async`: `async* listContainers`.  it would look symmetric with `yield*`.

Reply:
Prettier is doing this, not me.

Reply:
ok then, ignore this.